frontend/taipy-gui/src/components/Taipy/Chart.tsx

```diff
@@ -788,9 +861,89 @@ const Chart = (props: ChartProp) => {
                 }
             }
         },
-        [getRealIndex, dispatch, updateVars, propagate, props.onChange, config.traces.length, module]
+        [getRealIndex, dispatch, updateVars, propagate, props.onChange, config.traces.length, module],
     );
 
+    useEffect(() => {
+        if (!dataPl.length || !animationDataValues) return;
+
+        const toFramesData = dataPl.map((trace) => {
+            const isYaxisAnimated = Object.prototype.hasOwnProperty.call(animationDataValues, (trace as ScatterData).name);
+            const isXaxisAnimated = (trace as ExtendedPlotData).meta?.xAxisName &&
+                Object.prototype.hasOwnProperty.call(animationDataValues, (trace as ExtendedPlotData).meta?.xAxisName as PropertyKey);
+
+            return {
+                name: (trace as ScatterData).name,
+                x: isXaxisAnimated ? animationDataValues[(trace as ExtendedPlotData).meta?.xAxisName as keyof typeof animationDataValues] ?? [] : (trace as ScatterData).x,
+                y: isYaxisAnimated ? animationDataValues[(trace as ScatterData).name as keyof typeof animationDataValues] ?? [] : (trace as ScatterData).y,
+            };
+        });
+
+        setFrames((prevFrames) => ({
+            from: prevFrames.to
+                ? { ...prevFrames.to, name: "from" }
+                : { name: "from", data: [], traces: [], layout: {}, group: "", baseframe: "" },
+            to: {
+                name: "to",
+                data: toFramesData,
+                traces: dataPl.map((_, idx) => idx),
+                layout: {},
+                group: "",
+                baseframe: "from",
+            },
+        }));
+    }, [dataPl, animationDataValues]);
+
+    useEffect(() => {
+        if (!plotRef.current || !frames.to || !frames.to.data?.length) return;
+
+        const plotElement = plotRef.current as unknown as PlotlyHTMLElement;
+        if (!plotElement || !plotElement.data) return;
+
+        let animationTimeout: NodeJS.Timeout | null = null;
+
+        const addFramesAsync = async () => {
```

Comment:
```suggestion
        const animate= async () => {
```
Isn't exactly what this thing does?